lib/ansible/plugins/lookup/onepassword_doc.py

```diff
@@ -0,0 +1,125 @@
+      - This lookup can perform an initial login by providing C(subdomain), C(username), C(secret_key) and C(master_password).
+      - Due to the B(very) sensitive nature of these credentials, it is B(highly) recommended that you only pass in the minimal credentials
+        needed at any given time. Also, store these credentials in an Ansible Vault using a key that is equal to or greater in strength
+        to the 1Password master password.
+      - This lookup stores potentially sensitive data from 1Password as Ansible facts.
+        Facts are subject to caching if enabled, which means this data could be stored in clear text
+        on disk or in a database.
+      - Tested with C(op) version 0.5.3
+"""
+
+EXAMPLES = """
+# These examples only work when already signed in to 1Password
+- name: Retrieve document contents for MyAwesomeDocument when already signed in to 1Password
+  debug:
+    var: lookup('onepassword_doc', 'MyAwesomeDocument')
+
+- name: Retrieve document contents for MySecureTLSCert from a specific vault
+  debug:
+    var: lookup('onepassword_doc', 'MySecureTLSCert', vault='MyVault')
+
+- name: Retrieve MySecureTLSCert document when not signed in to 1Password
+  debug:
+    var: lookup('onepassword_doc'
+                'MySecureTLSCert'
+                subdomain='MySubdomain'
+                master_password=vault_master_password)
+
+- name: Retrieve MySecureTLSCert document when never signed in to 1Password
+  debug:
+    var: lookup('onepassword_doc'
+                'MySecureTLSCert'
+                subdomain='MySubdomain'
+                master_password=vault_master_password
+                username='tweety@acme.com'
+                secret_key=vault_secret_key)
+"""
+
+RETURN = """
+  _raw:
+    description: document contents requested
+"""
+
+from ansible.plugins.lookup.onepassword import OnePass
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils._text import to_bytes
+
+
+class OnePassDoc(OnePass):
+
+    def get_document_raw(self, doc_id, vault=None):
+        args = ["get", "document", doc_id]
+        if vault is not None:
+            args += ['--vault={0}'.format(vault)]
+        if not self.logged_in:
+            args += [to_bytes('--session=') + self.token]
+        rc, output, dummy = self._run(args)
+        return output
+
+    def get_document(self, doc_id, vault=None):
+        return self.get_document_raw(doc_id, vault)
+
+
+class LookupModule(LookupBase):
+
+    def run(self, doc_id, variables=None, **kwargs):
+        op = OnePassDoc()
+
+        vault = kwargs.get('vault')
+        op.subdomain = kwargs.get('subdomain')
+        op.username = kwargs.get('username')
+        op.secret_key = kwargs.get('secret_key')
+        op.master_password = kwargs.get('master_password', kwargs.get('vault_password'))
+
+        op.assert_logged_in()
+
+        return op.get_document(doc_id, vault)
```

Comment:
The method call to `run()` passes in a list terms, not a single string. If you only want to support a single term, you need to do some sanity checking here, and only pass in the first item in the list of terms.
```suggestion
        if len(doc_id) > 1:
            raise AnsibleLookupError('Only one item is supported')
        return op.get_document(doc_id[0], vault)
```

You could also just return each item in a list:

```suggestion
        docs = []
        for doc in doc_ids:
            docs.append(op.get_document(doc, vault))

        return docs
```

At a minimum you need to return the document as a single item list to avoid having each character in the document joined with a comma:
```suggestion
        return [op.get_document(doc_id, vault)]
```

You could accomplish this same thing by requiring the user to specify `wantlist=True` in the lookup call, but I think it is better to have it work nicely by default without having to specify that option.

This is the output I currently get with this lookup:
```yaml
tasks:
- debug:
    var: lookup('onepassword_doc', 'h5cjhlrtn5hg3pw2vihhwcmbcu')
  tags: debug
```

```
> ansible-playbook 1p-lookup.yml

PLAY [Test 1Password lookup] *****************************************************************

TASK [debug] *********************************************************************************
ok: [localhost] => {
    "lookup('onepassword_doc', 'h5cjhlrtn5hg3pw2vihhwcmbcu')": "-,-,-,-,-,B,E,G,I,N, ,R,S,A, ,P,R,I,V,A,T,E, ,K,E,Y,-,-,-,-,-

[output truncated]
```